tempodb/encoding/vparquet/compactor.go

```diff
@@ -206,3 +262,37 @@ func (b *bookmark) clear() {
 func (b *bookmark) close() {
 	b.iter.Close()
 }
+
+type rowPool struct {
+	pool    sync.Pool
+	maxSize int
+}
+
+func newRowPool(maxRowSize int) *rowPool {
+	return &rowPool{
+		maxSize: maxRowSize,
+		pool: sync.Pool{
+			New: func() any {
+				return make(parquet.Row, 0, maxRowSize)
+			},
+		},
+	}
+}
+
+func (r *rowPool) Get() parquet.Row {
+	return r.pool.Get().(parquet.Row)
+}
+
+func (r *rowPool) Put(row parquet.Row) {
+	if len(row) > r.maxSize {
+		return
+	}
+
+	// Clear before putting into the pool.
+	// This is important so that pool entries don't hang
+	// onto the underlying buffers.
+	for i := range row {
```

Comment:
does line 297 already do this by reducing the slice length to 0?

Reply:
I don't believe so.  String and []byte `parquet.Value`s contain a pointer to another buffer:
```go
type Value struct {
	ptr *byte
```

This loop erases the pointers to allow those buffers to be freed.  I believe the reference remains if only setting the slice len to 0.